tools/keytools/keygen.c

```diff
@@ -491,13 +492,54 @@ static void key_import(uint32_t ktype, const char *fname)
     uint8_t buf[KEYSLOT_MAX_PUBKEY_SIZE];
     FILE *f;
     int r;
+    int i;
+    int keySize = 0;
     f = fopen(fname, "rb");
     if (f == NULL) {
         fprintf(stderr, "Fatal error: could not open file %s to import public key\n", fname);
         exit(6);
     }
-    r = fread(buf, sizeof(buf), 1, f);
-    keystore_add(ktype, buf, r, fname);
+
+    switch (ktype) {
```

Comment:
This logic is duplicated above in `keystore_add`. Can this get consolidated to a single new function?

Reply:
Just this feedback item left.
Also can you investigate adding a test case for the import feature?

Reply:
yes the switch in keystore_add assumes raw keys and needs to be called from the key generate function but since import is now expected to handle keys with headers we need to check the length here as well, if it's just ecc keys that have this difference I can remove the switch and just do if statements for the 3 ecc key types

Reply:
I just mean a new static function for the duplicate code. There are two matching switch statements.

Reply:
ah you mean make a get key size function, thought you meant try to merge the two functions into a single since they had a similar part